streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -578,4 +577,10 @@ private StateStoreMetadata findStore(final TopicPartition changelogPartition) {
 
         return found.isEmpty() ? null : found.get(0);
     }
+
+    @Override
+    public TopicPartition changelogTopicPartitionFor(final String storeName) {
+        final StateStoreMetadata storeMetadata = stores.get(storeName);
+        return storeMetadata == null ? null : storeMetadata.changelogPartition;
```

Comment:
Cf. my comment from above: we could raise the exception if `storeMetadata == null`

Reply:
IMO, it is fine to return `null` here and mention both possible reason for the `null` result in `logChange()`. What exception would you throw here? Retrieving a changelog partition of a not registered state store is not exceptional. However, retrieving a changelog partition of a not registered state store in a method that writes to that changelog partition -- i.e. `logChange()` -- is exceptional.

Reply:
From my understanding `storeMetadata` should only be `null` if the store was not registered? Thus it seems to indicate a bug?

Reply:
I would say it indicates a bug, if the state store should be changelogged. My point is that retrieving a changelog partition for a state store does not necessarily mean that the state store needs to be registered. However, since `registeredChangelogPartitionFor()` is currently only called for changelogging, I will add two `IllegalStateException`s to this method: one for non registered stores and one for stores with logging disabled.